src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -0,0 +1,179 @@
+     *               this is stacked automatically by the VM.
+     * @param invocationName The invocation name, which is ignored.  When used with
+     *                       {@code invokedynamic}, this is provided by the
+     *                       {@code NameAndType} of the {@code InvokeDynamic}
+     *                       structure and is stacked automatically by the VM.
+     * @param invocationType The invocation type of the {@code CallSite}.  This
+     *                       method type should have a single parameter of
+     *                       a reference type, and return {@code int}.  When
+     *                       used with {@code invokedynamic}, this is provided by
+     *                       the {@code NameAndType} of the {@code InvokeDynamic}
+     *                       structure and is stacked automatically by the VM.
+     * @param labels non-null case labels - {@code String} and {@code Integer} constants
+     *                        and {@code Class} instances, in any combination
+     * @return a {@code CallSite}, which accepts two parameters: one is an instance
+     *         of the target type, and second is a restart index. It returns the
+     *         index into {@code labels} of the target value, if the target
+     *         is an instance of any of the types or equal to any of the constants, {@literal -1} if the target
+     *         value is {@code null}, or {@code types.length} if the target value
+     *         is not an instance of any of the types or equal to any of the constants. Will return
+     *         an index that is greater or equal to the restart index provided.
+     * @throws NullPointerException if any required argument is null
+     * @throws IllegalArgumentException if any labels are null, or if the
+     * invocation type is not {@code (T)int for some reference type {@code T}}
+     * @throws Throwable if there is any error linking the call site
+     */
+    public static CallSite typeSwitch(MethodHandles.Lookup lookup,
+                                      String invocationName,
+                                      MethodType invocationType,
+                                      Object... labels) throws Throwable {
+        if (invocationType.parameterCount() != 2
+            || (!invocationType.returnType().equals(int.class))
+            || invocationType.parameterType(0).isPrimitive())
+            throw new IllegalArgumentException("Illegal invocation type " + invocationType);
+        requireNonNull(labels);
+
+        labels = labels.clone();
+        Stream.of(labels).forEach(SwitchBootstraps::verifyLabel);
+
+        return new TypeSwitchCallSite(invocationType, labels);
+    }
+
+    private static void verifyLabel(Object label) {
+        if (Objects.isNull(label)) {
+            throw new IllegalArgumentException("null label found");
+        }
+        if (label.getClass() != Class.class &&
```

Comment:
store `label.getClass` in a local variable,
it's too bad that you can no use pattern matching here :)